src/hugr/views.rs

```diff
@@ -505,6 +590,19 @@ impl<T: AsRef<Hugr>> HugrView for T {
     }
 }
 
+/// Filter an iterator of node-ports to only dataflow dependency specifying
+/// ports (Value and StateOrder)
+pub fn dataflow_ports_only<'i, 'a: 'i, P: Into<Port> + Copy>(
```

Comment:
It would be a bit more practical to allow chained applications;

```rust
pub trait PortIterator {
    fn dataflow_ports_only(self, hugr) -> impl Iterator<...> { ... };
}
impl<I, P> PortIterator for I where I: Iterator<Item = (Node, P)> {}
```

But either way works alright.